crates/sui-core/src/execution_scheduler/balance_withdraw_scheduler/naive_scheduler.rs

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{collections::BTreeMap, sync::Arc};
+
+use parking_lot::Mutex;
+use sui_types::base_types::SequenceNumber;
+use tokio::sync::watch;
+
+use crate::execution_scheduler::balance_withdraw_scheduler::{
+    balance_read::AccountBalanceRead,
+    scheduler::{BalanceWithdrawSchedulerTrait, WithdrawReservations},
+    BalanceSettlement, ScheduleResult,
+};
+
+/// A naive implementation of the balance withdraw scheduler that does not attempt to optimize the scheduling.
+/// For each withdraw reservation, it will always wait until the dependent accumulator object is available,
+/// and then check if the balance is sufficient.
+/// This implementation is simple and easy to understand, but it is not efficient.
+/// It is only used to unblock further development of the balance withdraw scheduler.
+#[allow(dead_code)]
+pub(crate) struct NaiveBalanceWithdrawScheduler {
+    balance_read: Arc<dyn AccountBalanceRead>,
+    last_settled_version_sender: watch::Sender<SequenceNumber>,
+    last_settled_version_receiver: Mutex<Option<watch::Receiver<SequenceNumber>>>,
```

Comment:
Instead of holding the receiver in a Mutex it is probably better to make a new local receiver when you need one with `sender.subscribe()`